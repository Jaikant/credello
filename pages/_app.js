import { ThemeProvider } from 'styled-components';
import CssBaseline from '@material-ui/core/CssBaseline';
import { useEffect } from 'react';

export const theme = {
  colors: {
    primary: 'pink',
    grey: 'grey',
  },
};

function DefaultLayout({ children }) {
  return <>{children}</>;
}

function Credello({ Component, pageProps }) {
  useEffect(() => {
    const jssStyles = document.querySelector('#jss-server-side');
    if (jssStyles && jssStyles.parentNode)
      jssStyles.parentNode.removeChild(jssStyles);
  }, []);

  const Layout = Component.layout || DefaultLayout;
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Layout>
        <Component {...pageProps} />
      </Layout>
    </ThemeProvider>
  );
}

export default Credello;
