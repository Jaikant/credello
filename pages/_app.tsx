import {
  ThemeProvider,
  createGlobalStyle,
  DefaultTheme,
} from 'styled-components';
import CssBaseline from '@material-ui/core/CssBaseline';
import { useEffect } from 'react';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: "Roboto-Regular";
      src: url('/static/fonts/Roboto-Regular.ttf') format("truetype"),
  }
  @font-face {
    font-family: "Roboto-Medium";
      src: url('/static/fonts/Roboto-Medium.ttf') format("truetype"),
  }
  @font-face {
    font-family: "Roboto-Bold";
      src: url('/static/fonts/Roboto-Bold.ttf') format("truetype"),
  }
  @font-face {
    font-family: "Roboto-Black";
      src: url('/static/fonts/Roboto-Black.ttf') format("truetype"),
  }

  body {
    font-family: Roboto-Regular;
  }
  h1, h2, h3, h4, h5, h6 {
    font-family: Roboto-Bold;
  }
`;

export const theme: DefaultTheme = {
  colors: {
    primary: '#02254d',
    grey: 'grey',
  },
  spacing: {
    //top level paddings
    //Mobile
    paddingX: '22px',
    paddingTopHeroY: '40px',
    paddingBottomY: '32px',

    //top level paddings
    //tablet
    tabletPaddingX: '24px',
    tabletPaddingY: '40px',

    //top level paddings
    //laptop
    laptopPaddingX: '128px',
    laptopPaddingY: '40px',

    //component spacing
    padding12: '12px',
    padding13: '13px',
    padding16: '16px',
    padding22: '22px',
    padding28: '28px',
    padding30: '30px',
    padding32: '32px',
    padding36: '36px',
    padding48: '48px',
  },
};

function DefaultLayout({ children }) {
  return <>{children}</>;
}

function Credello({ Component, pageProps }) {
  useEffect(() => {
    const jssStyles = document.querySelector('#jss-server-side');
    if (jssStyles && jssStyles.parentNode)
      jssStyles.parentNode.removeChild(jssStyles);
  }, []);

  const Layout = Component.layout || DefaultLayout;
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <GlobalStyle />
      <Layout>
        <Component {...pageProps} />
      </Layout>
    </ThemeProvider>
  );
}

export default Credello;
