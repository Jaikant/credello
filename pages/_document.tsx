import Document, {
  DocumentContext,
  Html,
  Head,
  Main,
  NextScript,
} from 'next/document';
import { ServerStyleSheet } from 'styled-components';
import { ServerStyleSheets as MaterialUiServerStyleSheets } from '@material-ui/core/styles';

export default class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const styledComponentSheet = new ServerStyleSheet();
    const materialUiSheets = new MaterialUiServerStyleSheets();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            styledComponentSheet.collectStyles(
              materialUiSheets.collect(<App {...props} />)
            ),
        });

      const initialProps = await Document.getInitialProps(ctx);
      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            {materialUiSheets.getStyleElement()}
            {styledComponentSheet.getStyleElement()}
          </>
        ),
      };
    } finally {
      styledComponentSheet.seal();
    }
  }
  render() {
    return (
      <Html lang="en">
        <Head>
          <link
            rel="preload"
            href="/static/fonts/Roboto-Medium.ttf"
            as="font"
            crossOrigin=""
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
