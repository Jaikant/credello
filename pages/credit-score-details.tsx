import Layout from 'components/Layout';
import { CredelloFC } from 'libs/types';
import BreadCrumbs from 'components/BreadCrumbs';
import Stepper from 'components/Stepper';

export const DebtAccounts: CredelloFC = () => (
  <div>
    <BreadCrumbs mb="18px" />
    <Stepper active={3} mb="30px" />
    <div> Debt accounts ... </div>
  </div>
);

DebtAccounts.layout = Layout;
export default DebtAccounts;
