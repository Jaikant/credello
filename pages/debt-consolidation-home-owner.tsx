import { useContext } from 'react';
import Layout from 'components/Layout';
import Link from 'next/link';
import { CredelloFC } from 'libs/types';
import BreadCrumbs from 'components/BreadCrumbs';
import { Text, TextBold } from 'components/Typography';
import Stepper from 'components/Stepper';
import BottomBar from 'components/BottomBar';
import { MainContext } from 'libs/context';
import { YesOrNoButtons } from 'components/Buttons';

export const HomeOwner: CredelloFC = () => {
  const context = useContext(MainContext);
  const updateHomeOwner = (value) =>
    context.dispatch({ type: 'homeOwner', value });

  return (
    <div>
      <BreadCrumbs mb="18px" />
      <Stepper active={4} mb="30px" />
      <TextBold color="#02254d" lh="22px">
        If you are a homeowner, you might have the option to consolidate with
        mortgage refinancing or home equity loans- which generally have much
        lower interest rates.
      </TextBold>
      <TextBold color="#02254d" mb="40px" lh="22px">
        Do you own a home?
      </TextBold>

      <YesOrNoButtons
        fn={updateHomeOwner}
        initial={context.appState.homeOwner}
      />
      {context && context.appState.homeOwner && (
        <Text size="14px" lh="20px" mt="32px">
          <Link href="#">
            <Text as="span" size="14px" lh="20px" color="#14cc8c">
              {`Skip home section `}
            </Text>
          </Link>
          if you don&apos;t want to explore home equity options
        </Text>
      )}

      <BottomBar
        left="/debt-consolidation-income"
        right={
          context.appState && context.appState.homeOwner
            ? '/home-owner'
            : '/404'
        }
        position="fixed"
      />
    </div>
  );
};

HomeOwner.layout = Layout;
export default HomeOwner;
