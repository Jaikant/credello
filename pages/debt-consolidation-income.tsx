import React, { useContext } from 'react';
import Layout from 'components/Layout';
import { CredelloFC } from 'libs/types';
import BreadCrumbs from 'components/BreadCrumbs';
import { TextBold } from 'components/Typography';
import Stepper from 'components/Stepper';
import BottomBar from 'components/BottomBar';
import Question from 'components/Question';
import { MainContext } from 'libs/context';
import { YesOrNoButtons } from 'components/Buttons';

export const MonthlyIncome: CredelloFC = () => {
  const context = useContext(MainContext);
  const updatePayOffAbility = (value) =>
    context.dispatch({ type: 'payoffAbility', value });
  return (
    <div>
      <BreadCrumbs mb="18px" />
      <Stepper active={3} mb="30px" />
      <TextBold color="#02254d" mb="16px" lh="22px">
        What is your monthly income?{' '}
      </TextBold>

      <TextBold color="#02254d" mb="16px" lh="22px">
        Will you be able to payoff $xxx within the next 18 or 21 months?
      </TextBold>
      <YesOrNoButtons
        fn={updatePayOffAbility}
        initial={context.appState.payoffAbility}
      />

      <BottomBar
        left="/credit-score-details"
        right="/debt-consolidation-home-owner"
        position="fixed"
      />
    </div>
  );
};

MonthlyIncome.layout = Layout;
export default MonthlyIncome;
