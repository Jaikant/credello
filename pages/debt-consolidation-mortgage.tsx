import { useRef } from 'react';
import Layout from 'components/Layout';
import Link from 'next/link';
import styled from 'styled-components';
import { CredelloFC } from 'libs/types';
import BreadCrumbs from 'components/BreadCrumbs';
import { Text, TextBold, TextMedium } from 'components/Typography';
import Stepper from 'components/Stepper';
import BottomBar from 'components/BottomBar';
import FormMortgageDetails from 'components/FormMortgageDetails';

export const DebtAccounts: CredelloFC = () => {
  const submitRef = useRef<HTMLFormElement | null>();

  return (
    <div>
      <BreadCrumbs mb="18px" />
      <Stepper active={4} mb="30px" />
      <TextBold color="#02254d" mb="16px" lh="22px">
        You are almost there!
      </TextBold>
      <TextBold color="#02254d" mb="16px" lh="22px">
        Finally, tell us about your current mortgage loan, so we can assess if
        consolidating using home equity might work for you.
      </TextBold>
      <FormMortgageDetails submitRef={submitRef} />
      <BottomBar
        left="/home-owner"
        submit={() => {
          submitRef.current.dispatchEvent(
            new Event('submit', { cancelable: true })
          );
        }}
        position="fixed"
      />
    </div>
  );
};

DebtAccounts.layout = Layout;
export default DebtAccounts;
