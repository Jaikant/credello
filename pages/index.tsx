import Layout from 'components/Layout';
import styled from 'styled-components';
import { CredelloFC } from 'libs/types';
import Header from 'components/Header';
import CompareProducts from 'components/CompareProducts';
import PopularCalculators from 'components/PopularCalculators';
import FeaturedArticles from 'components/FeaturedArticles';
import ArticleGrid from 'components/ArticleGrid';
import SectionWithButton from 'components/SectionWithButton';
import Footer from 'components/Footer';

const StyledWrapper = styled.div``;

export const Home: CredelloFC = () => (
  <>
    <Header />
    <CompareProducts />
    <PopularCalculators />
    <FeaturedArticles />
    <StyledWrapper>
      <ArticleGrid />
      <SectionWithButton />
      <Footer />
    </StyledWrapper>
  </>
);

Home.layout = Layout;
export default Home;
