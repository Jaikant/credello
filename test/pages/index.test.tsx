import React from 'react';
import { render } from '../testUtils';
import { Home } from '../../pages/index';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    // const { asFragment } = render(<Footer />, {});
    // expect(asFragment()).toMatchSnapshot();

    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });

  // it('clicking button triggers alert', () => {
  //   const { getByText } = render(<Home />, {})
  //   window.alert = jest.fn()
  //   fireEvent.click(getByText('Test Button'))
  //   expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
});
