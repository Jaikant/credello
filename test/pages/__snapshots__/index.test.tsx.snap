// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
.c2 {
  font-family: Roboto;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.2;
  webkit-font-smoothing: antialiased;
  margin: 0;
  margin-bottom: 32px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c1 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 32px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c3 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c4 {
  display: grid;
  grid-gap: 8px;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr 1fr;
}

.c7 {
  background-color: #5cdb95;
  background-image: url('test-file-stub');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  padding: 24px;
  border-radius: 5px;
  color: #fff;
  font-weight: 700;
  line-height: 1.2;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
}

.c6 {
  background-color: #5cdb95;
  background-image: url('test-file-stub');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  padding: 24px;
  border-radius: 5px;
  color: #fff;
  font-weight: 700;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
}

.c8 {
  background-color: #5cdb95;
  background-image: url('test-file-stub');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  padding: 24px;
  border-radius: 5px;
  color: #fff;
  font-weight: 700;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
}

.c5 {
  background-color: #5cdb95;
  background-image: url('test-file-stub');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  padding: 24px;
  border-radius: 5px;
  color: #fff;
  font-weight: 700;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
}

.c10 {
  font-family: Roboto;
  font-weight: 500;
  font-size: 12px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 0;
  margin-top: 8px;
  color: #5cdb95;
  text-align: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  margin: 8px;
}

.c0 {
  background-image: url('test-file-stub');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: top right;
  margin: -32px -16px;
  padding: 32px 16px;
}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c2 {
    font-size: 16px;
  }
}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {
  .c1 {
    font-size: 32px;
  }
}

@media (min-width:1224px) {
  .c1 {
    line-height: 1.5;
    font-size: 32px;
  }
}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c3 {
    line-height: 1.5;
  }
}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c4 {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 12px;
  }
}

@media (min-width:1224px) {
  .c10 {
    font-size: 12px;
  }
}

@media (min-width:768px) {
  .c9 {
    margin: 16px;
  }
}

@media (min-width:1224px) {
  .c9 {
    margin: 32px;
  }
}

@media (min-width:768px) {
  .c0 {
    display: grid;
    grid-gap: 32px;
    background-image: url('test-file-stub');
    background-size: 50% 100%,contain;
    margin: -32px -24px;
    padding: 32px 24px;
  }
}

@media (min-width:1224px) {
  .c0 {
    display: grid;
    grid-gap: 64px;
    background-image: url('test-file-stub');
    background-size: 50% auto,contain;
    margin: -32px -128px;
    padding: 32px 128px;
  }
}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

<div
  class="c0"
>
  <div>
    <div
      class="c1"
    >
      Make financial decisions simple
    </div>
    <div
      class="c2"
    >
      with personalized guidance
    </div>
  </div>
  <div>
    <div
      class="c3"
    >
      Manage your debt better
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        Ways to pay off debt
      </div>
      <div
        class="c6"
      >
        Estimate debt-free date
      </div>
      <div
        class="c7"
      >
        Ways to consolidate
      </div>
      <div
        class="c8"
      >
        Estimate savings
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="Typography__TextMedium-sc-12f2dd0-3 c10"
        color="#5cdb95"
      >
        Learn more about debt consolidation
      </div>
    </div>
  </div>
</div>
`;
